isa/kernel.org/instruction-set.rst

```diff
@@ -318,49 +446,221 @@ the atomics features, while keeping a lower ``-mcpu`` version, you can use
 64-bit immediate instructions
 -----------------------------
 
-Instructions with the ``BPF_IMM`` mode modifier use the wide instruction
+Instructions with the ``BPF_IMM`` 'mode' modifier use the wide instruction
 encoding for an extra imm64 value.
 
 There is currently only one such instruction.
 
 ``BPF_LD | BPF_DW | BPF_IMM`` means::
 
-  dst_reg = imm64
+  dst = imm64
 
 
 Legacy BPF Packet access instructions
 -------------------------------------
 
 eBPF has special instructions for access to packet data that have been
 carried over from classic BPF to retain the performance of legacy socket
-filters running in the eBPF interpreter.
+filters running in an eBPF interpreter.
 
 The instructions come in two forms: ``BPF_ABS | <size> | BPF_LD`` and
 ``BPF_IND | <size> | BPF_LD``.
 
 These instructions are used to access packet data and can only be used when
-the program context is a pointer to networking packet.  ``BPF_ABS``
+the program context contains a pointer to a networking packet.  ``BPF_ABS``
 accesses packet data at an absolute offset specified by the immediate data
 and ``BPF_IND`` access packet data at an offset that includes the value of
 a register in addition to the immediate data.
 
 These instructions have seven implicit operands:
 
- * Register R6 is an implicit input that must contain pointer to a
-   struct sk_buff.
+ * Register R6 is an implicit input that must contain a pointer to a
+   context structure with a packet data pointer.
  * Register R0 is an implicit output which contains the data fetched from
    the packet.
- * Registers R1-R5 are scratch registers that are clobbered after a call to
-   ``BPF_ABS | BPF_LD`` or ``BPF_IND | BPF_LD`` instructions.
+ * Registers R1-R5 are scratch registers that are clobbered by the
+   instruction.
 
-These instructions have an implicit program exit condition as well. When an
-eBPF program is trying to access the data beyond the packet boundary, the
-program execution will be aborted.
+*Linux implementation note*: In Linux, R6 references a struct sk_buff.
+
+These instructions have an implicit program exit condition as well. If an
+eBPF program attempts access data beyond the packet boundary, the
+program execution must be gracefully aborted.
 
 ``BPF_ABS | BPF_W | BPF_LD`` means::
 
-  R0 = ntohl(*(u32 *) (((struct sk_buff *) R6)->data + imm32))
+  R0 = ntohl(*(uint32_t *) (R6->data + imm))
+
+where `ntohl()` converts a 32-bit value from network byte order to host byte order.
 
 ``BPF_IND | BPF_W | BPF_LD`` means::
 
-  R0 = ntohl(*(u32 *) (((struct sk_buff *) R6)->data + src_reg + imm32))
+  R0 = ntohl(*(uint32_t *) (R6->data + src + imm))
+
+Appendix
+========
+
+For reference, the following table lists opcodes in order by value.
```

Comment:
We probably need some mark-up around the descriptions to mark them as litteral, as the `*` and `|` are currently interpreted as RST mark-up.

Reply:
Thanks.  Github still renders them correctly (see https://github.com/dthaler/ebpf-docs/blob/new/isa/kernel.org/instruction-set.rst#appendix) so I hadn't noticed.

Reply:
It's not entirely clear to me why it adds links to some of the `*`, to be honest

Reply:
Ok, I've fixed the asterisks now